### ISSUE TITLE
PD013 UseOfDotStack triggered by np.stack

### ISSUE THREAD

**Issue by ManonMarchand** [charliermarsh/ruff#1659]
Hi, 

I think this is not intentional: using the numpy function `stack` with `np.stack` triggers the error PD013 `UseOfDotStack .melt is preferred to .stack; provides same functionality` even if this has nothing to do with a Pandas dataframe.

**Comment by charliermarsh** [comment]
Yeah we can probably guard against this.

**Comment by charliermarsh** [comment]
(We could ensure that whatever’s referenced isn’t an in import. Maybe some other heuristics too.)

**Comment by twoertwein** [comment]
(same for `torch.stack`)

**Comment by ManonMarchand** [comment]
Is there a way to check the type of what's referenced? The issue might also occur in all the other `UseOfDot*` checks

**Comment by edgarrmondragon** [comment]
> Is there a way to check the type of what's referenced?

Not really, since types would have to be tracked beyond the module being analyzed. The best we could do is check the imports, like Charlie said above.

**Comment by charliermarsh** [comment]
Yeah, I think we can at least ensure that `np.stack` and `torch.stack` themselves don't trigger. What we can't guard against are things like:

```py
x = np.array([])
x.stack(...)
```

(That's not valid NumPy code anyway, more illustrating the point that this would be a false positive.)
